runtime/hsa-runtime/core/runtime/amd_topology.cpp

```diff
@@ -68,58 +72,42 @@ extern r_debug _amdgpu_r_debug;
 
 namespace rocr {
 namespace AMD {
-// Minimum acceptable KFD version numbers
-static const uint kKfdVersionMajor = 0;
-static const uint kKfdVersionMinor = 99;
-
-void DiscoverDrivers(bool &gpu_found, bool &aie_found) {
-  // Open connection to GPU and AIE kernel drivers.
-  gpu_found = (KfdDriver::DiscoverDriver() == HSA_STATUS_SUCCESS);
-  aie_found = (XdnaDriver::DiscoverDriver() == HSA_STATUS_SUCCESS);
-}
+// Anonymous namespace.
+namespace {
+const std::vector<std::function<hsa_status_t(std::unique_ptr<core::Driver> &)>>
```

Comment:
You can use `std::array` here and avoid the allocation during static initialization (until we have constexpr containers that is).

Reply:
Fixed